apps/meet-app/backend/service.bal

```diff
@@ -75,122 +78,357 @@ service http:InterceptableService / on new http:Listener(9090) {
-        if cacheError is error {
-            log:printError(string `Error in updating the user cache for: ${userInfo.email}!`);
+        // Fetch the user's privileges based on the roles
+        int[] privileges = [];
+        if authorization:checkPermissions([authorization:authorizedRoles.SALES_TEAM],
+                userInfo.groups) {
+            privileges.push(authorization:SALES_TEAM_PRIVILEGE);
+        }
+        if authorization:checkPermissions([authorization:authorizedRoles.SALES_ADMIN],
+                userInfo.groups) {
+            privileges.push(authorization:SALES_ADMIN_PRIVILEGE);
+        }
+
+        return <http:Ok>{
+            body: {
+                ...loggedInUser,
+                privileges
+            }
+        };
+    }
+
+    # Fetch meeting types from the database.
+    #
+    # + domain - Domain to filter 
+    # + return - Meeting types | Error
+    isolated resource function get meetings/types(http:RequestContext ctx, string domain)
+        returns database:MeetingTypes|http:Forbidden|http:InternalServerError {
+
+        // User information header
+        authorization:CustomJwtPayload|error userInfo = ctx.getWithType(authorization:HEADER_USER_INFO);
+        if userInfo is error {
+            return <http:InternalServerError>{
+                body: {
+                    message: "User information header not found!"
+                }
+            };
+        }
+
+        // Custom Resource level authorization.
+        if !authorization:checkPermissions([authorization:authorizedRoles.SALES_TEAM], userInfo.groups) {
+            return <http:Forbidden>{
+                body: {
+                    message: "Insufficient privileges!"
+                }
+            };
+        }
+
+        // Fetch the meeting types from the database
+        database:MeetingTypes|error? meetingTypes = database:fetchMeetingTypes(domain);
+        if meetingTypes is error {
+            string customError = string `Error occurred while retrieving the meeting types!`;
+            log:printError(customError, meetingTypes);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+        if meetingTypes is null {
+            string customError = string `No meeting types found for the given domain!`;
+            log:printError(customError);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+
+        return meetingTypes;
+    }
+
+    # Create a new meeting.
+    #
+    # + createCalendarEventRequest - Create calendar event request
+    # + return - Created meeting | Error
+    isolated resource function post meetings(
+            http:RequestContext ctx, calendar:CreateCalendarEventRequest createCalendarEventRequest)
+        returns http:Created|http:Forbidden|http:InternalServerError {
```

Comment:
```suggestion
    isolated resource function post meetings(http:RequestContext ctx,
        calendar:CreateCalendarEventRequest createCalendarEventRequest)
        returns http:Created|http:Forbidden|http:InternalServerError {
```

Reply:
We might need to manually handle the indentation in the `returns` line as the formatter doesn't provide that